### DISCUSSION TITLE
Templates for provisioners stored in db

### DISCUSSION THREAD

**Discussion by PavanKumar-MSK** [smallstep/certificates#1825] [Q&A] [answered]
How can I add templates to the provisioners defined in the db?
I looked at this open issue but could not understand how to do this - 
[Issue ](https://github.com/smallstep/cli/issues/720)

**Comment by tashian** [comment, marked as answer]
You can use `--x509-template=file` and provide a [template file](https://smallstep.com/docs/step-ca/templates/) as input.

There is also a `--x509-template-data=file` which allows you to set variables that will be used in the template of the provisioner. This is helpful if you to share use one template across many provisioners, but with subtle changes to one or more values.

Does this help?

**Reply by PavanKumar-MSK** [reply]
Ok. I am able to see that under **Options** [here](https://smallstep.com/docs/step-cli/reference/ca/provisioner/update/#usage). I was a bit confused as the command under **Usage** does not have that flag.
```
step ca provisioner update <name> [--public-key=<file>]
[--private-key=<file>] [--create] [--password-file=<file>]
[--admin-cert=<file>] [--admin-key=<file>] [--admin-subject=<subject>]
[--admin-provisioner=<name>] [--admin-password-file=<file>]
[--ca-url=<uri>] [--root=<file>] [--context=<name>] [--ca-config=<file>]
```
